x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/AbstractPyTorchAction.java

```diff
@@ -65,12 +65,18 @@ public final void init() {
     }
 
     void onTimeout() {
+        onTimeout(new ElasticsearchStatusException("timeout [{}] waiting for inference result", RestStatus.REQUEST_TIMEOUT, timeout));
+    }
+
+    void onCancel() {
+        onTimeout(new ElasticsearchStatusException("inference task cancelled", RestStatus.INTERNAL_SERVER_ERROR));
```

Comment:
Like you say, the client won't ever see the response, hence there's no appropriate status code for this.

I've committed your suggestion.

Reply:
One last thought: I think 5xx makes more sense than 4xx here. 4xx's are client errors, meaning you should fix your request. However, there's nothing wrong with the request here. It's a server-side problem.

`504 Gateway Timeout` probably makes most sense, since ES acts as a gateway to PyTorch, which times out.

Anyway, since this never reaches the client, I'll just leave it as is.